### DISCUSSION TITLE
Lesson 5 infura issue

### DISCUSSION THREAD

**Discussion by leopashov** [smartcontractkit/full-blockchain-solidity-course-py#81] [Q&A] [answered]
I'm trying to deploy the simple storage contract to rinkeby using brownie (I'm around the 4hrs 58 mark) but wanted to try the option of using my own wallet through brownie (the second of the three options, which allows us to set a password)
As a result, I have changed the 'get_account' function to:

```
def get_account():
    if network.show_active() == "development":
        return accounts[0]
    else:
        return accounts.load("wallet_name")
```

and my .env file only contains one line:
`export WEB3_INFURA_PROJECT_ID = my_infura_project_id`

when running `brownie run scripts/deploy.py --network rinkeby`

I get:
`HTTPError: 401 Client Error: Unauthorized for url: https://rinkeby.infura.io/v3/my_infura_project_id`

Does anyone have any advice?
Thank you!

**Comment by ChakibHi** [comment]
Two possibilities:
1- "`wallet_name`" not properly defined in the `brownie-config` file.
2- your `infura_id` is not properly set in the `.env`file.

I found that @PatrickAlphaC answered a similar issue on Stackexchange.
link: https://ethereum.stackexchange.com/a/106854

**Reply by leopashov** [reply]
Turned out I hadn't made a brownie config file.
Now I do, all that's in it is:
`dotenv: .env`
how do I properly define the "wallet_name" in the config file? 
It seems defined by brownie - brownie asks for the password for the wallet when I run the script, it also shows up in `brownie accounts list`
Patrick's answer to the other issue seems pretty focused on the '3rd' method of doing this (ie having the private key in the .env file) which I'm trying to avoid

**Reply by leopashov** [reply]
I have also just realised the error actually shows:
`HTTPError: 401 Client Error: Unauthorized for url: https://rinkeby.infura.io/v3/0xmy_infura_project_id`
despite the .env file containing:
`export WEB3_INFURA_PROJECT_ID=my_infura_project_id`
I have tried resaving, deleting the .env file and remaking it, rerunning etc. but could this be the issue?
Have also copied the 'method 3' option of passing private key (as described in the stackexchange answer) and am getting the same result

**Reply by ChakibHi** [reply]
If so, no need to define the wallet_name in config file.
I suggest to redefine the network again (to override the previous one), this time the infura_ID is properly set. 
Edit:
delete the 0x from your saved infura_ID.

**Reply by leopashov** [reply]
sorry what do you mean by redefining the network? Where is this done?

**Reply by leopashov** [reply]
I have deleted the 0x from saved infura id (and saved the .env file) but the error is still the same - ie error still shows 0xinfura_id

**Comment by AlbertoSinigaglia** [comment, marked as answer]
pretty sure to had one of those errors weeks ago, and it was because you should not use spaces in exports, so instead of
```
export WEB3_INFURA_PROJECT_ID = my_infura_project_id
```
try 
```
export WEB3_INFURA_PROJECT_ID=my_infura_project_id
```

**Reply by leopashov** [reply]
Tried this but didn't fix it - will defo keep doing this in future though thank you

**Reply by AlbertoSinigaglia** [reply]
the infura token on my pc is `export WEB3_INFURA_PROJECT_ID=d4dd9edaXXXXXXXXXXXXXXXXXXXXXXXX` without any `0x` and spaces

**Reply by leopashov** [reply]
I've done the same now but the .env file doesn't seem to be changing anything
I have setup a new project in infura and taken that project ID and saved the .env file but the error still gives the old project id in the hyper link.
Is there some kind of hidden memory vs code is using when I use brownie run...?

**Reply by AlbertoSinigaglia** [reply]
@leopashov just delete the network from brownie and add again

**Reply by leopashov** [reply]
It seems I've managed to delete the network but can't figure out how to add it back.
I've managed to add one called:

environment=[Ethereum (Infura)]
  └─id=[Rinkeby (Infura)]: id=[Rinkeby (Infura)]

in brownie networks list
which I now can't get rid of (because I can't figure out how to use this syntax)
which was also the reason it's named like that 
could you please help with what I'm meant to be typing into command line and the rules around syntax? It would help me a lot

**Reply by AlbertoSinigaglia** [reply]
You can find the commands in the video (idk at which point) but the command looks like this:
```
brownie networks add development something-fork cmd=ganache-cli host=http://127.0.0.1 fork=https://mainnet.infura.io/v3/XXXXXXX  accounts=10 mnemonic=brownie port=8545

**Reply by leopashov** [reply]
Yay got it all working now thank you!
command I used to re add rinkeby chain was:
`brownie networks add Ethereum rinkeby host=https://rinkeby.infura.io/v3/My_infura_project_id 
chainid=4`
